src/rviz_visual_tools.cpp

```diff
@@ -971,14 +971,18 @@ bool RvizVisualTools::publishCone(const geometry_msgs::Pose& pose, double angle,
 bool RvizVisualTools::publishABCDPlane(const double A, const double B, const double C, const double D,
                                         colors color, double x_width, double y_width)
 {
-  Eigen::Isometry3d pose;
+  if (D > 0)
+  {
+    ROS_WARN_STREAM_NAMED(LOGNAME, "For a plane equation Ax+By+Cz+D=0, D should be <=0.");
```

Comment:
why? that's not true, the normal vector can have a 'negative' direction.

Reply:
It took me awhile to get this straight. If you look at rviz_visual_tools_demo.cpp, there is:

Ax+By+D=0

So D == -Ax-By

Let's say a point on the plane is (1,1). Then A = 1, B = 1, D = -1-1 = -2

The radius from (0,0) to (1,1) is sqrt(-D) -- in other words, positive D causes issues.

I think this is a little confusing because PCL uses Ax+By+Cz+D=0. Usually I see Ax+By+Cz=D^2 or Ax+By+Cz-D=0

Reply:
Let's decide to use Ax+By+Cz+D=0 (which to my knowledge is most common).

> The radius from (0,0) to (1,1) is sqrt(-D) -- in other words, positive D causes issues.

what radius are you talking about? The sign of D defines in what direction the normal vector has to point from the origin towards the center point of the plane (and therefore defines the location of the center point). The distance of the plane towards the origin is |D| / |(A,B,C)| so we don't have any problems with negative or positive signs, both are clearly defined.

The only issue I see here is that we need to document what version of ABCD we're using (namely positive or negative D).

Reply:
I'd just follow this definition http://mathworld.wolfram.com/Plane.html and compute distance and normal vector as described